app/src/main/java/org/septa/android/app/transitview/TransitViewFragment.java

```diff
@@ -0,0 +1,290 @@
+package org.septa.android.app.transitview;
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.septa.android.app.R;
+import org.septa.android.app.TransitType;
+import org.septa.android.app.database.DatabaseManager;
+import org.septa.android.app.domain.RouteDirectionModel;
+import org.septa.android.app.support.CursorAdapterSupplier;
+import org.septa.android.app.support.RouteModelComparator;
+import org.septa.android.app.view.TextView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class TransitViewFragment extends Fragment implements TransitViewLinePickerFragment.TransitViewLinePickerListener {
+
+    private static final String TAG = TransitViewFragment.class.getSimpleName();
+
+    private RouteDirectionModel firstRoute, secondRoute, thirdRoute;
+    private CursorAdapterSupplier<RouteDirectionModel> busRouteCursorAdapterSupplier, trolleyRouteCursorAdapterSupplier;
+    public static final String TRANSITVIEW_ROUTE_FIRST = "TRANSITVIEW_ROUTE_FIRST",
+            TRANSITVIEW_ROUTE_SECOND = "TRANSITVIEW_ROUTE_SECOND",
+            TRANSITVIEW_ROUTE_THIRD = "TRANSITVIEW_ROUTE_THIRD";
+
+    // layout variables
+    private TextView firstRoutePicker, secondRoutePicker, thirdRoutePicker;
+    private View queryButton;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+
+        if (getActivity() == null) {
```

Comment:
Given everything you are doing in this CreateView method I would get a reference to Activity and use that in place of all your getActivity() calls.

Reply:
resolved with https://github.com/septadev/SEPTA-Android/pull/182/commits/3f51f08517c75ed3e5f810f4acec66d277c0f313